x/ccv/provider/keeper/throttle.go

```diff
@@ -322,16 +323,22 @@ func (k Keeper) QueueThrottledPacketData(
 	case ccvtypes.SlashPacketData:
 		bz, err = data.Marshal()
 		if err != nil {
+			// Slash packet data is assumed to be validated in OnRecvPacket and/or OnRecvSlashPacket,
```

Comment:
`QueueThrottledPacketData` is called only when receiving a packet (in DeliverTX). Thus, it's cleaner to return an error and send back an err ack if something goes wrong.

Reply:
Agreed 👍, my previous understanding was that "this should never happen" types of marshaling errors should always panic, as it's assumed calling methods handle validation. Just confirming we're okay moving away from that pattern?

Reply:
I've addressed this comment in https://github.com/cosmos/interchain-security/commit/8f5bd9462f269d262fddd14c9f31e807257b1c82. It's notable that returning errors like this does introduce more boilerplate. We should eventually settle on a standardized pattern for external data that needs to be validated